specs/predeploys.md

```diff
@@ -26,62 +29,84 @@ Predeployed smart contracts exist on Optimism at predetermined addresses in
+| LegacyMessagePasser           | 0x4200000000000000000000000000000000000000 | Legacy     | Yes        | Yes     |
+| DeployerWhitelist             | 0x4200000000000000000000000000000000000002 | Legacy     | Yes        | Yes     |
+| LegacyERC20ETH                | 0xDeadDeAddeAddEAddeadDEaDDEAdDeaDDeAD0000 | Legacy     | Yes        | No      |
+| WETH9                         | 0x4200000000000000000000000000000000000006 | Legacy     | No         | Yes     |
+| L2CrossDomainMessenger        | 0x4200000000000000000000000000000000000007 | Legacy     | No         | Yes     |
+| L2StandardBridge              | 0x4200000000000000000000000000000000000010 | Legacy     | No         | Yes     |
+| SequencerFeeVault             | 0x4200000000000000000000000000000000000011 | Legacy     | No         | Yes     |
+| OptimismMintableERC20Factory  | 0x4200000000000000000000000000000000000012 | Legacy     | No         | Yes     |
+| L1BlockNumber                 | 0x4200000000000000000000000000000000000013 | Legacy     | Yes        | Yes     |
+| GasPriceOracle                | 0x420000000000000000000000000000000000000F | Legacy     | No         | Yes     |
+| GovernanceToken               | 0x4200000000000000000000000000000000000042 | Legacy     | No         | No      |
+| L1Block                       | 0x4200000000000000000000000000000000000015 | Bedrock    | No         | Yes     |
+| L2ToL1MessagePasser           | 0x4200000000000000000000000000000000000016 | Bedrock    | No         | Yes     |
+| L2ERC721Bridge                | 0x4200000000000000000000000000000000000014 | Legacy     | No         | Yes     |
+| OptimismMintableERC721Factory | 0x4200000000000000000000000000000000000017 | Bedrock    | No         | Yes     |
+| ProxyAdmin                    | 0x4200000000000000000000000000000000000018 | Bedrock    | No         | No      |
+| BaseFeeVault                  | 0x4200000000000000000000000000000000000019 | Bedrock    | No         | Yes     |
+| L1FeeVault                    | 0x420000000000000000000000000000000000001a | Bedrock    | No         | Yes     |
+
+## LegacyMessagePasser
+
+[Implementation](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/contracts/legacy/LegacyMessagePasser.sol)
+
+Address: `0x4200000000000000000000000000000000000000`
+
+The `LegacyMessagePasser` contract stores commitments to withdrawal
+transactions before the Bedrock upgrade. A merkle proof to a particular
+storage slot that commits to the withdrawal transaction is used as part
+of the withdrawing transaction on L1. The expected account that includes
+the storage slot is hardcoded into the L1 logic. After the bedrock upgrade,
+the `L2ToL1MessagePasser` is used instead. Finalizing withdrawals from this
+contract will no longer be supported after the Bedrock and is only left
+to allow for alternative bridges that may depend on it.
```

Comment:
So what happens if you try to use post-Bedrock, does it still work (e.g. by forwarding to `L2ToL1MessagePasser`)?